src/nitro/communication/messages/parser/handshake/DisconnectReasonParser.ts

```diff
@@ -0,0 +1,30 @@
+import { IMessageDataWrapper, IMessageParser } from '../../../../../api';
+
+export class DisconnectReasonParser implements IMessageParser
+{
+    private _reason: number;
+
+    public flush(): boolean
+    {
+        this._reason = -1;
+
+        return true;
+    }
+
+    public parse(wrapper: IMessageDataWrapper): boolean
+    {
+        if(!wrapper) return false;
+
+        if(wrapper.bytesAvailable)
```

Comment:
initialize `this._reason` to `0` of no bytes available. This is how ActionScript behaves (would implicitly initialize integers to 0)